test/spec/services/apiServiceSpec.js

```diff
@@ -234,6 +235,96 @@ describe("APIService", function(){
+    it('should return list of kinds that are scoped to a namespace by default', function() {
+      var namespacedKinds = _.map(APIService.availableKinds(), 'kind');
+      // from the fixtures included, we should have 51 namespaced kinds
+      expect( _.difference(bothSample, namespacedKinds).length ).toEqual(0);
+    });
+
+    it('should return list of all kinds, including those that are cluster scoped, when passed a truthy argument', function() {
+      var allKinds = _.map(APIService.availableKinds(true), 'kind');
+      // from the fixtures included, we should have 84 total kinds
+      expect( _.difference(bothSample, allKinds).length ).toEqual(0);
+      expect( _.difference(onlyClusterSample, allKinds).length ).toEqual(0);
+    });
+
+    // kinds from the old /oapi should not be iterated at all.
+    it('should not list kinds from the old /oapi namespace (that do not have a group)', function() {
+      var allKinds = APIService.availableKinds(true);
+      var shouldNotBeFound = [];
+      // This is a sampling of items from /oapi that should no longer be listed
+      var oapiShouldNotExistSample = [
+        {kind: 'ClusterPolicy'},
+        {kind: 'ClusterRole'},
+        {kind: 'Image'},
+        {kind: 'Template'},
+        {kind: 'Project'},
+        {kind: 'User'}
+      ];
+      _.each(oapiShouldNotExistSample, function(kindToFind) {
+        var found = _.find(allKinds, function(kind) {
+          return (kind.kind === kindToFind.kind) && !_.includes(_.keys(kind), 'group');
+        });
+        if(found) {
+          shouldNotBeFound.push(found);
+        }
+      });
+      expect(shouldNotBeFound.length).toEqual(0);
+    });
+
+    // unlike the /oapi endpoint, the /api endpoint should still be listed
+    it('should list items from the k8s /api namespace (that do not have a group)', function() {
+      var allKinds = APIService.availableKinds(true);
+      var shouldBeFound = [];
+      // this is a sampling of items from /api that should still be listed,
+      // even though they do not yet have a group associated.
+      var k8sAPIStillExistsSample = [
+        {kind: 'Binding'},
+        {kind: 'ConfigMap'},
+        {kind: 'Namespace'},
+        {kind: 'PersistentVolume'},
+        {kind: 'Service'},
+        {kind: 'ServiceAccount'},
+        {kind: 'Pod'}
+      ];
+      _.each(k8sAPIStillExistsSample, function(kindToFind) {
+        var found = _.find(allKinds, function(kind) {
+          return (kind.kind === kindToFind.kind) && !_.includes(_.keys(kind), 'group');
+        });
+        if(found) {
+          shouldBeFound.push(found);
+        }
+      });
+      expect(shouldBeFound.length).toEqual(7);
+    });
+
+    // When a kind is promoted from openshift to kube, for a time, we will allow it to exist in both api groups
+    it('should not return duplicate kinds, except for those we have promoted from openshift to kube', function() {
```

Comment:
i would eliminate this test altogether, its just not true.  there may always be duplicate kinds when we have aggregated api servers doing any number of things, i think your other two tests confirm the right thing, which is that we no longer include oapi in the list of available kinds, but that we do still include /api